src/game/AI/ScriptDevAI/scripts/world/npcs_special.cpp

```diff
@@ -2805,6 +2805,206 @@ bool GossipHello_npc_gossip_npc(Player* player, Creature* creature)
+            pPlayer->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "Jewelcrafting", GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 5);
+            pPlayer->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "Alchemy", GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 6);
+            pPlayer->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "Blacksmithing", GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 7);
+            pPlayer->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "Enchanting", GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 8);
+            pPlayer->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "Engineering", GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 9);
+            pPlayer->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "Herbalism", GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 10);
+            pPlayer->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "Leatherworking", GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 11);
+            pPlayer->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "Mining", GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 12);
+            pPlayer->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "Skinning", GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 13);
+            pPlayer->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "Tailoring", GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 14);
+            pPlayer->ADD_GOSSIP_ITEM(GOSSIP_ICON_CHAT, "First Aid", GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 15);
+            pPlayer->SEND_GOSSIP_MENU(39288, pCreature->GetObjectGuid());
+            return true;
+        case GOSSIP_ACTION_INFO_DEF + 2:
+            factionEntry = sFactionStore.LookupEntry<FactionEntry>(932);
+            pPlayer->GetReputationMgr().ModifyReputation(factionEntry, 84000);
+            pPlayer->CLOSE_GOSSIP_MENU();
+            return true;
+        case GOSSIP_ACTION_INFO_DEF + 3:
+            factionEntry = sFactionStore.LookupEntry<FactionEntry>(934);
+            pPlayer->GetReputationMgr().ModifyReputation(factionEntry, 84000);
+            pPlayer->CLOSE_GOSSIP_MENU();
+            return true;
+        case GOSSIP_ACTION_INFO_DEF + 4:
+            factionEntry = sFactionStore.LookupEntry<FactionEntry>(967);
+            pPlayer->GetReputationMgr().ModifyReputation(factionEntry, 84000);
+            factionEntry = sFactionStore.LookupEntry<FactionEntry>(990);
+            pPlayer->GetReputationMgr().ModifyReputation(factionEntry, 84000);
+            factionEntry = sFactionStore.LookupEntry<FactionEntry>(1012);
+            pPlayer->GetReputationMgr().ModifyReputation(factionEntry, 84000);
+            pPlayer->CLOSE_GOSSIP_MENU();
+            return true;
+        case GOSSIP_ACTION_INFO_DEF + 5:
+            if (pPlayer->GetFreePrimaryProfessionPoints() > 0)
+            {
+                pPlayer->learnSpellHighRank(28897);
+                pPlayer->SetSkill(755, 375, 375);
+                LearnSkillRecipesHelper(pPlayer, 755);
+            }
+            pPlayer->CLOSE_GOSSIP_MENU();
+            return true;
+        case GOSSIP_ACTION_INFO_DEF + 6:
+            if (pPlayer->GetFreePrimaryProfessionPoints() > 0)
```

Comment:
Shouldnt this check be instead around actually showing the options rather than selecting them?

Reply:
Logically yes, on the reference however the options always show and just don't allow you to learn more professions if you already have 2